DER/DerivativesContracts/RightsAndWarrants.rdf

```diff
@@ -105,7 +113,57 @@
 	<owl:Class rdf:about="&fibo-der-drc-raw;AllotmentRightFormula">
 		<rdfs:subClassOf rdf:resource="&fibo-fnd-utl-alx;Formula"/>
 		<rdfs:label xml:lang="en">allotment right formula</rdfs:label>
-		<skos:definition xml:lang="en">formula used to calculate the number of securities for an allotment right, typically based on the number of these instruments that the holder holds</skos:definition>
+		<skos:definition xml:lang="en">rule applied to calculate the number of securities for an allotment right, typically based on the number of these instruments that the holder holds</skos:definition>
+		<fibo-fnd-utl-av:explanatoryNote xml:lang="en">Note that there may be a combination of a rule expressed in text as well as an expression or more complex formula embedded in a contract for making this determination.</fibo-fnd-utl-av:explanatoryNote>
+	</owl:Class>
+	
+	<owl:Class rdf:about="&fibo-der-drc-raw;BasketWarrant">
+		<rdfs:subClassOf rdf:resource="&fibo-der-drc-raw;Warrant"/>
+		<rdfs:subClassOf>
+			<owl:Restriction>
+				<owl:onProperty rdf:resource="&fibo-der-drc-bsc;hasUnderlier"/>
+				<owl:someValuesFrom rdf:resource="&fibo-sec-sec-bsk;BasketOfSecurities"/>
+			</owl:Restriction>
+		</rdfs:subClassOf>
+		<rdfs:label xml:lang="en">basket warrant</rdfs:label>
+		<skos:definition xml:lang="en">warrant that permits the holder to acquire a package or group of assets at a specified price during a specified period</skos:definition>
+		<fibo-fnd-utl-av:adaptedFrom xml:lang="en">ISO 10962, Securities and related financial instruments - Classification of financial instruments (CFI) code, Fourth Edition, October 2019</fibo-fnd-utl-av:adaptedFrom>
+	</owl:Class>
+	
+	<owl:Class rdf:about="&fibo-der-drc-raw;BondPurchaseRight">
+		<rdfs:subClassOf rdf:resource="&fibo-der-drc-raw;PurchaseRight"/>
+		<rdfs:subClassOf>
+			<owl:Restriction>
+				<owl:onProperty rdf:resource="&fibo-der-drc-bsc;hasUnderlier"/>
+				<owl:someValuesFrom>
+					<owl:Restriction>
+						<owl:onProperty rdf:resource="&fibo-fnd-rel-rel;hasIdentity"/>
+						<owl:someValuesFrom rdf:resource="&fibo-sec-dbt-bnd;Bond"/>
+					</owl:Restriction>
+				</owl:someValuesFrom>
+			</owl:Restriction>
+		</rdfs:subClassOf>
+		<rdfs:label xml:lang="en">bond purchase right</rdfs:label>
+		<skos:definition xml:lang="en">purchase right that gives a prospective acquiree&apos;s bondholders the right to buy bonds of the firm or bonds of anyone who acquires the firm at a deep discount to their fair market value</skos:definition>
```

Comment:
ambiguous - needs commas around "or bonds of anyone who acquires the firm". 
Though "firm" seems over specific and not an official FIBO term.
More generally I could not find "bond purchase right" used anywhere - is it really a thing? And is it really only limited to a company aquisition?